crates/tap-agent/src/agent/sender_allocation.rs

```diff
@@ -1262,19 +1269,149 @@ pub mod tests {
+        mock_server
+            .register(
+                Mock::given(method("POST"))
+                    .and(body_string_contains("transactions"))
+                    .respond_with(
+                        ResponseTemplate::new(200)
+                            .set_body_json(json!({ "data": { "transactions": []}})),
+                    ),
+            )
+            .await;
+
+        populate(pgpool.clone()).await;
+
+        let (mut message_receiver, sender_account) = create_mock_sender_account().await;
+
+        // Create a sender_allocation.
+        let (sender_allocation, notify) = create_sender_allocation()
+            .pgpool(pgpool.clone())
+            .sender_aggregator_endpoint("http://".to_owned() + &aggregator_endpoint.to_string())
+            .escrow_subgraph_endpoint(&mock_server.uri())
+            .rav_request_receipt_limit(2000)
+            .sender_account(sender_account)
+            .call()
+            .await;
+
+        // Trigger a RAV request manually and wait for updated fees.
+        sender_allocation
+            .cast(SenderAllocationMessage::TriggerRAVRequest)
+            .unwrap();
+
+        flush_messages(&notify).await;
+
+        let total_unaggregated_fees = call!(
+            sender_allocation,
+            SenderAllocationMessage::GetUnaggregatedReceipts
+        )
+        .unwrap();
+
+        // Check that the unaggregated fees are correct.
+        assert_eq!(total_unaggregated_fees.value, 0u128);
+
+        let startup_msg = message_receiver.recv().await.unwrap();
+        let expected_value: u128 = ((amount_of_receipts.to_u128().expect("should be u128")
+            - 1u128)
+            * (amount_of_receipts.to_u128().expect("should be u128")))
+            / 2;
+        assert_eq!(
+            startup_msg,
+            SenderAccountMessage::UpdateReceiptFees(
+                ALLOCATION_ID_0,
+                ReceiptFees::UpdateValue(UnaggregatedReceipts {
+                    value: expected_value,
+                    last_id: amount_of_receipts,
+                    counter: amount_of_receipts,
+                })
+            )
+        );
+
+        // Stop the TAP aggregator server.
+        handle.stop().unwrap();
+        handle.stopped().await;
+    }
+
+    #[sqlx::test(migrations = "../../migrations")]
+    async fn test_several_receipts_rav_request(pgpool: PgPool) {
+        const AMOUNT_OF_RECEIPTS: u64 = 1000;
+        execute(
+            pgpool,
+            |pgpool| async move {
+                // Add receipts to the database.
+
+                for i in 0..AMOUNT_OF_RECEIPTS {
+                    let receipt =
+                        create_received_receipt(&ALLOCATION_ID_0, &SIGNER.0, i, i + 1, i.into());
+                    store_receipt(&pgpool, receipt.signed_receipt())
+                        .await
+                        .unwrap();
+                }
+            },
+            AMOUNT_OF_RECEIPTS,
+        )
+        .await;
```

Comment:
This is pretty nice, really easy to understand what's the diff between those two tests.